source/guide_tomcat.rst

```diff
@@ -0,0 +1,156 @@
+ [isabell@stardust ~]$ which java
+ /usr/bin/java
+ [isabell@stardust ~]$
+
+Also check your URL and, if you like, set up :manual_anchor:`additional domains <web-domains>`:
+
+.. include:: includes/web-domain-list.rst
+
+
+Installation
+============
+
+Create application directory and files
+--------------------------------------
+
+First create a folder called ``tomcat``:
+
+::
+
+ [isabell@stardust ~]$ mkdir ~/tomcat
+ [isabell@stardust ~]$
+
+
+Next, download the `current version <https://tomcat.apache.org/download-10.cgi>`_ of Tomcat and unpack it into your new folder:
+
+::
+
+ [isabell@stardust ~]$ wget https://dlcdn.apache.org/tomcat/tomcat-10/v10.0.12/bin/apache-tomcat-10.0.12.tar.gz
+ Resolving dlcdn.apache.org (dlcdn.apache.org)... 151.101.2.132, 2a04:4e42::644
+ Connecting to dlcdn.apache.org (dlcdn.apache.org)|151.101.2.132|:443... connected.
+ HTTP request sent, awaiting response... 200 OK
+ Length: 11890640 (11M) [application/x-gzip]
+ Saving to: ‘apache-tomcat-10.0.12.tar.gz’
+ [isabell@stardust ~]$ tar --extract --file=apache-tomcat-10.0.12.tar.gz --directory=tomcat/ --strip-components=1
+ [isabell@stardust ~]$
+
+
+Configuration
+==============
+
+Set up service
+--------------
+
+Create the service file ``~/etc/services.d/tomcat.ini`` and fill it with:
+
+::
+
+ [program:tomcat]
+ environment=CATALINA_HOME=%(ENV_HOME)s/tomcat
+ command=%(ENV_HOME)s/tomcat/bin/catalina.sh run
+
+You could explicitely set further environment variables like ``JAVA_HOME`` (required for debugging) or ``CATALINA_BASE`` (specify directories for multiple instances). These are documented in the initial comment block of ``~/tomcat/bin/catalina.sh`` but are not required for a minimal installation.
+
+.. include:: includes/supervisord.rst
+
+For your convenience, you may also want to create a symbolic link to the log file. Note the differences between ``catalina.out`` and ``catalina.YYYY-MM-DD.log`` explained `here <https://stackoverflow.com/questions/51985958/what-is-the-difference-between-catalina-out-and-catalina-yyyy-mm-dd-log-log>`_ .
+
+::
+
+ [isabell@stardust ~]$ ln -s ~/tomcat/logs/catalina.out ~/logs/tomcat.log
+
+
+Tomcat is now up and running as a service. To make it accessible from the outside, we need to configure a web backend.
+
+
+
+Set up web backend
+------------------
+
+.. note::
+
+    Tomcat is running on port 8080. This is only relevant for the server but can nevertheless be changed in ``~/tomcat/conf/server.xml``.
+
+.. include:: includes/web-backend.rst
+
+
+At this point, Tomcat should be visible at ``https://isabell.uber.space``. However, to really make use of the management interface you still need to give yourself access.
+
+
+Set up web management users
+---------------------------
+
+To make the management interface usable, edit ``~/tomcat/conf/tomcat-users.xml``. You will find several blocks of comments that you may want to delete or simply uncomment. Make sure to add this line and set an appropriate password:
```

Comment:
I think its not totally clear here what  your mean with "this line". Because there already is a `<tomcat-users ...>` that should be used. And one should only put the `<user ...` line within this container.